### PR TITLE
Review

### PR DESCRIPTION
**Pull request by enemtsev**

### CODE REVIEW

File: src/wheelMenuModel.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        emit startSaveConfig();
+    });
+
+    m_saveThread->start();
+    m_configSaveTimer->start(5000);
+}
+
+WheelMenuModel::~WheelMenuModel()
+{
+    m_saveThread->quit();
```
Comment (enemtsev) [inline src/wheelMenuModel.cpp:51]: is some dtors missing here? or, probably, connections in ctor?

File: src/wheelMenuModel.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    int innerRowIndex() const;
+    void setInnerRowIndex(int newInnerRowIndex);
+
+signals:
+    void innerRowIndexChanged();
+    void startSaveConfig();
+
+private:
+    std::vector<MenuItemPtr> m_menuItems;
```
Comment (enemtsev) [inline src/wheelMenuModel.h:64]: could you please explain the purpose of using unique_ptr to store data in the vector?

File: src/wheelMenuModel.cpp (hunk 1 above)
Comment (Palm1r) [reply src/wheelMenuModel.cpp:51]: yeah, my mistake. I need to add
```suggestion
    m_saveThread->quit();
    m_saveThread->wait();
```
and add before
    connect(m_saveThread, &QThread::finished, m_saveThread, &QThread::deleteLater);
    connect(m_saveThread, &QThread::finished, m_saveWorker, &QObject::deleteLater);

File: src/wheelMenuModel.h (hunk 2 above)
Comment (Palm1r) [reply src/wheelMenuModel.h:64]: In simple models, using unique_ptr makes it easier to work with object lifetimes. Later, I saw copy problem here. Maybe using simple types make it easier here

Comment (Palm1r) [reply src/wheelMenuModel.h:64]: Although I think using QVector with MenuItem and COW would be great here too

File: src/fileSaver.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+FileSaver::FileSaver()
+    : m_pingPong{&m_menuItemsCopyPing, &m_menuItemsCopyPong}
+{}
+
+void FileSaver::copyData(ModelData modelData)
+{
+    int writeIndex = m_currentIndex == 0 ? 1 : 0;
+
+    std::vector<std::unique_ptr<MenuItem>> *writeBuffer = m_pingPong[writeIndex];
+
```
Comment (Palm1r) [inline src/fileSaver.cpp:15]: Necessary to add vector<>::reserve

File: src/wheelMenuModel.cpp (hunk 1 above)
Comment (enemtsev) [reply src/wheelMenuModel.cpp:51]: :+1:

File: src/wheelMenuModel.h (hunk 2 above)
Comment (enemtsev) [reply src/wheelMenuModel.h:64]: :+1:

File: src/fileSaver.cpp (hunk 3 above)
Comment (enemtsev) [reply src/fileSaver.cpp:15]: :+1:

**Comment by enemtsev** [comment]
Thank you for your efforts. Not much to discuss here :-) I like the result and provided comments. 
From my side I am recommending to say "yes" to our HR. There are some other candidates, so overall answer might be not immediate.

**Pull request closed without merging**